ui/component/or-asset-tree/src/index.ts

```diff
@@ -538,6 +568,99 @@ export class OrAssetTree extends subscribe(manager)(LitElement) {
         this._onNodeClicked(null, null);
     }
 
+    /**
+     * On Cancel button click, we clear everything and reset the asset tree to original content
+     *
+     * @param event
+     * @protected
+     */
+    protected _onCancelFilterChange() {
+        this._filterValue = '';
+        this._foundAssets = [];
+
+        this._loadAssets();
+    }
+
+    protected _onFilterValueChange(e: OrInputChangedEvent) {
+        this._filterValue = e.detail.value;
+        const oldFilterValue: string = e.detail.previousValue;
+
+        if (!this._filterValue.startsWith(oldFilterValue) || this._filterValue.length === 0) {
+            this._onCancelFilterChange();
+        }
+
+        if (this._searchInputTimer) {
+            clearTimeout(this._searchInputTimer);
+        }
+
+        this._searchInputTimer = setTimeout(async () => {
+            if (this._nodes && this._filterValue.length > 2) {
+                let newNodes: UiAssetTreeNode[] = this._nodes;
+                this._nodes = [];
+                this._isFiltering = true;
+
+                const query: AssetQuery = {
+                    select: {
+                        excludePath: true,
+                        excludeAttributes: true,
+                        excludeParentInfo: true
+                    },
+                    names: [{
+                        predicateType: "string",
+                        match: AssetQueryMatch.CONTAINS,
+                        value: this._filterValue,
+                        caseSensitive: false
+                    }]
+                };
+
+                const response = await manager.rest.api.AssetResource.queryAssets(query);
+
+                this._foundAssets = response.data.map<string>((asset: Asset) => asset.id ? asset.id : '' );
```

Comment:
If state becomes part of the `UiAssetTreeNode` as mentioned above then I don't see a need to store `_foundAssets`

Reply:
In previous comment, I succeed to get rid of this :-)